### PR TITLE
[FMT] format example/02_decoding

### PR DESCRIPTION
**Pull request by Remi-Gau**
Relates to https://github.com/nilearn/nilearn/issues/2528

Changes proposed in this pull request:

- apply pre-commit hooks to example/02_decoding

### Files involved in other PRs:

- 'examples/02_decoding/plot_haxby_frem.py',
- 'examples/02_decoding/plot_haxby_searchlight_surface.py',

### CODE REVIEW

File: examples/02_decoding/plot_haxby_searchlight_surface.py
Comment (NicolasGensollen) [inline examples/02_decoding/plot_haxby_searchlight_surface.py:67]: I don't understand why this import statement was moved here... It should be below the comment (which will be parsed as the section title), right ?

File: examples/02_decoding/plot_miyawaki_encoding.py
Comment (NicolasGensollen) [inline examples/02_decoding/plot_miyawaki_encoding.py:175]: Same here

File: examples/02_decoding/plot_miyawaki_reconstruction.py
Comment (NicolasGensollen) [inline examples/02_decoding/plot_miyawaki_reconstruction.py:27]: Should be below comment. But I don't think this comment is useful at all, so we might be better removing the comment instead...

File: examples/02_decoding/plot_haxby_searchlight.py
Comment (Remi-Gau) [reply examples/02_decoding/plot_haxby_searchlight.py:58]: Not sure but defo a black thing

```bash
black --diff tmp.py
--- tmp.py      2023-03-02 10:07:52.701237 +0000
+++ tmp.py      2023-03-02 10:07:54.130716 +0000
@@ -1 +1 @@
-process_mask[..., (picked_slice + 1):] = 0
+process_mask[..., (picked_slice + 1) :] = 0
would reformat tmp.py
```

File: examples/02_decoding/plot_miyawaki_reconstruction.py
Comment (NicolasGensollen) [inline examples/02_decoding/plot_miyawaki_reconstruction.py:153]: Same here, the comment only restates what the import statement is already saying.

File: examples/02_decoding/plot_miyawaki_reconstruction.py
Comment (Remi-Gau) [reply examples/02_decoding/plot_miyawaki_reconstruction.py:122]: Same

```bash
black --diff --color tmp.py
--- tmp.py      2023-03-02 10:10:05.445856 +0000
+++ tmp.py      2023-03-02 10:10:19.336303 +0000
@@ -1,3 +1,3 @@
 
-height_tf = (np.eye(y_cols) + np.eye(y_cols, k=1))[:y_cols - 1] * .5
+height_tf = (np.eye(y_cols) + np.eye(y_cols, k=1))[: y_cols - 1] * 0.5
would reformat tmp.py
```

File: examples/02_decoding/plot_oasis_vbm.py
Comment (NicolasGensollen) [inline examples/02_decoding/plot_oasis_vbm.py:46]: Same, I think we want this import below the comment

File: examples/02_decoding/plot_oasis_vbm.py
Comment (NicolasGensollen) [inline examples/02_decoding/plot_oasis_vbm.py:193]: If we think this comment is really useful, then we should move it to the line just above now

File: examples/02_decoding/plot_oasis_vbm_space_net.py
Comment (NicolasGensollen) [inline examples/02_decoding/plot_oasis_vbm_space_net.py:29]: should be below comment

**Comment by ymzayek** [comment]
Quick comment, before merging this I think we should make sure other PRs are not open on the example files. For example #3483 is working on a decoding example. We should try to minimize introducing conflicts in contributors PRs as was planned with our formatting strategy. (This goes for all the example formatting PRs of course though I haven't checked if there are other potential conflicts)

**Review by NicolasGensollen — COMMENTED** [review]
Thanks for working on this @Remi-Gau !
Just a few suggestions but LGTM otherwise !

**Comment by Remi-Gau** [comment]
> Quick comment, before merging this I think we should make sure other PRs are not open on the example files. For example #3483 is working on a decoding example. We should try to minimize introducing conflicts in contributors PRs as was planned with our formatting strategy. (This goes for all the example formatting PRs of course though I haven't checked if there are other potential conflicts)

So just to explain how those changes were done to try to minimize merge conflicts:

1. I set up the precommit config to not touch any files that were in any PR: see this comment (https://github.com/nilearn/nilearn/pull/3548/files#r1122405725)
2. Then used `pre-commit run -a` to autoformat as much as possible.
3. Applied flynt to convert to f-strings "en masse"
4. Fixed manually any remaining issues (flake8 mostly)
5. Applied small refactoring suggested by sourcery
6. Only committed the files for a given example folder (had tried to them all in on PR but it would have hell to review)
7. The PR [for example/07_advanced](https://github.com/nilearn/nilearn/pull/3556) is a WIP because it also contains the CI changes to check all the examples folders: this PR will have to merged last.

**Comment by ymzayek** [comment]
@Remi-Gau thanks for the detailed explanation! All sounds good to me in that case :)

**Comment by Remi-Gau** [comment]
> @Remi-Gau thanks for the detailed explanation! All sounds good to me in that case :)

I think that it will be hard to get 0% merge conflict through the reformatting process.

Those examples PR may have a couple of changes on files involved somewhere else.

Will try to flag those whenever I can

**Comment by Remi-Gau** [comment]
> @Remi-Gau thanks for the detailed explanation! All sounds good to me in that case :)

@ymzayek I upgraded the script I used to list the files in PR to also tell us the files that are in several PR.

For each formatting PR I have added a list of the files that are involved in at least one other PR: this way we can try to minimize how much change is done on those files.

File: examples/02_decoding/plot_haxby_full_analysis.py
Comment (Remi-Gau) [reply examples/02_decoding/plot_haxby_full_analysis.py]: should be fixed

File: examples/02_decoding/plot_haxby_different_estimators.py
Comment (Remi-Gau) [reply examples/02_decoding/plot_haxby_different_estimators.py]: should be fixed

File: examples/02_decoding/plot_miyawaki_reconstruction.py
Comment (Remi-Gau) [reply examples/02_decoding/plot_miyawaki_reconstruction.py]: should be fixed

**Review by bthirion — APPROVED** [review]
LGTM

File: examples/02_decoding/plot_haxby_searchlight_surface.py
Comment (Remi-Gau) [reply examples/02_decoding/plot_haxby_searchlight_surface.py]: I have tried to move the import in sensible place and re ran isort to make sure it was happy with it.

File: examples/02_decoding/plot_miyawaki_encoding.py
Comment (Remi-Gau) [reply examples/02_decoding/plot_miyawaki_encoding.py]: I have tried to move the import in sensible place and re ran isort to make sure it was happy with it.

File: examples/02_decoding/plot_miyawaki_reconstruction.py
Comment (Remi-Gau) [reply examples/02_decoding/plot_miyawaki_reconstruction.py:25]: did that

File: examples/02_decoding/plot_oasis_vbm.py
Comment (Remi-Gau) [reply examples/02_decoding/plot_oasis_vbm.py]: done

File: examples/02_decoding/plot_miyawaki_reconstruction.py
Comment (Remi-Gau) [reply examples/02_decoding/plot_miyawaki_reconstruction.py]: done

File: examples/02_decoding/plot_oasis_vbm_space_net.py
Comment (Remi-Gau) [reply examples/02_decoding/plot_oasis_vbm_space_net.py]: done

**Review by ymzayek — APPROVED** [review]
Ok to merge once the CI passes. Thx @Remi-Gau

**Pull request merged**
